src/generators/clientFileGenerator.ts

```diff
@@ -594,27 +594,28 @@ function getTrack2DefaultContent(
     return (
       defaultContent +
       `
+      let bearerTokenAuthenticationPolicyFound: boolean = false;
       if (options?.pipeline && options.pipeline.getOrderedPolicies().length > 0) {
         const pipelinePolicies: coreRestPipeline.PipelinePolicy[] = options.pipeline.getOrderedPolicies();
-        const bearerTokenAuthenticationPolicyFound = pipelinePolicies.some(
+        bearerTokenAuthenticationPolicyFound = pipelinePolicies.some(
           pipelinePolicy =>
             pipelinePolicy.name ===
             coreRestPipeline.bearerTokenAuthenticationPolicyName
         );
-        if (!bearerTokenAuthenticationPolicyFound) {
-          this.pipeline.removePolicy({
-            name: coreRestPipeline.bearerTokenAuthenticationPolicyName
-          });
-          this.pipeline.addPolicy(
-            coreRestPipeline.bearerTokenAuthenticationPolicy({
-              scopes: \`\${optionsWithDefaults.baseUri}/.default\`,
-              challengeCallbacks: {
-                authorizeRequestOnChallenge:
-                  coreClient.authorizeRequestOnClaimChallenge
-              }
-            })
-          );
-        }
+      }
+      if (!options || !options.pipeline || options.pipeline.getOrderedPolicies().length == 0 || !bearerTokenAuthenticationPolicyFound) {
```

Comment:
@sarangan12 Could you double-confirm the second issue I mentioned in thread?
Otherwise this fix will break all authentications of all new clients once released

Reply:
@MaryGao This was added only for the second issue. 

Let us consider the user does not pass anything in the option, then the options will be null. So, Policy will be set. 

If the User sets the options, but if pipeline is null, the policy will be set. 

If the Pipeline is set, but there are no policies, the policy will be set. 

If the Pipeline has policy, but not the bearer token authentication policy, then the new policy will be set.

Reply:
@sarangan Linked the second issue [here](https://github.com/Azure/azure-sdk-for-js/issues/21820)

Yes, I could totally understand your change here. 

However I will recommend to fix the second issue before than merging the your pr to codegen, otherwise once this policy is applied it will fail auth for new clients.